modules/apps/frontend-js/frontend-js-spa-web/test/app/App.ts

```diff
@@ -2145,102 +2383,114 @@ describe('App', function () {
-		this.app.addSurfaces(['surfaceId1']);
-		this.app.navigate('/path1#surfaceId1').then(() => {
-			const surfaceNode = document.querySelector('#surfaceId1');
+		app = new App();
+
+		app.addRoutes(new Route('/path1', Screen));
+		app.addSurfaces([new Surface('surfaceId1')]);
+
+		app.navigate('/path1#surfaceId1').then(() => {
+			const surfaceNode = document.querySelector(
+				'#surfaceId1'
+			) as HTMLElement;
+
+			if (!surfaceNode) {
+				return;
+			}
+
 			const {offsetLeft, offsetTop} = getNodeOffset(surfaceNode);
+
 			expect(window.pageYOffset).toBe(offsetTop);
 			expect(window.pageXOffset).toBe(offsetLeft);
-			document.getElementById('element').remove();
+
+			document.getElementById('element')?.remove();
+
 			done();
 		});
 	});
 
 	it('updates the document.referrer upon navigation', (done) => {
-		this.app = new App();
-		this.app.addRoutes(new Route('/path1', Screen));
-		this.app.addRoutes(new Route('/path2', Screen));
-		this.app.addRoutes(new Route('/path3', Screen));
+		app = new App();
+		app.addRoutes(new Route('/path1', Screen));
+		app.addRoutes(new Route('/path2', Screen));
+		app.addRoutes(new Route('/path3', Screen));
 
-		this.app
-			.navigate('/path1')
+		app.navigate('/path1')
 			.then(() => {
-				return this.app.navigate('/path2');
+				return app?.navigate('/path2');
 			})
 			.then(() => {
 				expect(getUrlPathWithoutHash(document.referrer)).toBe('/path1');
 
-				return this.app.navigate('/path3');
+				return app?.navigate('/path3');
 			})
 			.then(() => {
 				expect(getUrlPathWithoutHash(document.referrer)).toBe('/path2');
-				this.app.on(
-					'endNavigate',
-					() => {
-						expect(getUrlPathWithoutHash(document.referrer)).toBe(
-							'/path1'
-						);
-						done();
-					},
-					true
-				);
+				app?.on('endNavigate', () => {
+					expect(getUrlPathWithoutHash(document.referrer)).toBe(
+						'/path1'
+					);
+					done();
+				});
 				window.history.back();
 			});
 	});
 
 	it('does reload page on navigate back to a routed page without history state and skipLoadPopstate is active', () => {
-		this.app = new App();
-		this.app.reloadPage = jest.fn();
-		this.app.addRoutes(new Route('/path1', Screen));
-		this.app.addRoutes(new Route('/path2', Screen));
+		app = new App();
+
+		app.reloadPage = jest.fn();
 
-		return this.app.navigate('/path1').then(() => {
-			window.history.replaceState(null, null, null);
+		app.addRoutes(new Route('/path1', Screen));
+		app.addRoutes(new Route('/path2', Screen));
 
-			return this.app.navigate('/path2').then(() => {
+		return app.navigate('/path1').then(() => {
+			window.history.replaceState(null, '', null);
+
+			return app?.navigate('/path2').then(() => {
 				return new Promise((resolve) => {
 					window.addEventListener(
 						'popstate',
 						() => {
-							expect(this.app.reloadPage).not.toHaveBeenCalled();
+							expect(app?.reloadPage).not.toHaveBeenCalled();
+
 							resolve();
 						},
```

Comment:
`Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?`

`resolve()` expects an argument, not sure what to give it here.